core/src/main/scala/com/rallyhealth/vapors/core/time/CountTime.scala

```diff
@@ -0,0 +1,44 @@
+package com.rallyhealth.vapors.core.time
+
+import java.time.temporal.{ChronoUnit, Temporal, TemporalUnit}
+
+trait CountTime[-T, -U] {
+
+  /**
+    * Count the whole number of complete time units between the start and end.
+    *
+    * @param start the start of the time range (inclusive)
+    * @param end the end of the time range (inclusive)
+    * @param truncateToUnit the unit to round down to
+    * @return the whole number of units between the start up to, but not including, the end
+    */
+  def between(
+    start: T,
+    end: T,
+    truncateToUnit: U,
+  ): Long
+}
+
+object CountTime {
+
+  def between[T, U](
+    start: T,
+    end: T,
+    roundDownToUnit: U,
+  )(implicit
+    diffTime: CountTime[T, U],
+  ): Long = diffTime.between(start, end, roundDownToUnit)
+
+  private final class OfTemporal[-T <: Temporal] extends CountTime[T, TemporalUnit] {
+    override def between(
+      start: T,
+      end: T,
+      unit: TemporalUnit,
+    ): Long = start.until(end, unit)
+  }
+
+  @inline private def temporal[T <: Temporal]: CountTime[T, ChronoUnit] = new OfTemporal[T]
+
+  // TODO: Restrict invalid units by type
+  implicit val ofTemporal: CountTime[Temporal, ChronoUnit] = temporal
```

Comment:
There isn't a good type distinction between `ChronoUnit.SECONDS` and `ChronoUnit.YEARS` or `ChronoUnit.FOREVER`, but it will blow up at runtime if you attempt to count the amount of time in years or "forever"s (whatever that means) between `Instant`s. This would require creating my own units that map to other time units, but maintain compile-time safety. I will take a crack at this when I have more time.